### PR TITLE
Theano backend cudnn implementation of batch normalization added

### PR DESCRIPTION
**Pull request by shashankg7**
As suggested by @nouiz, added bn implementation from cudnn based on availability of gpu.

### CODE REVIEW

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -324,8 +325,11 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, std, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, std, beta and gamma.
     '''
-    normed = T.nnet.bn.batch_normalization(x, gamma, beta, mean, std + epsilon,
-                                           mode='high_mem')
+    if theano.config.device.startswith('gpu'):
+        normed = dnn_batch_normalization_test(x, gamma, beta, mean, std ** 2, epsilon)
```
Comment (fchollet) [inline keras/backend/theano_backend.py]: Wy `**2`?

Have you look at the `mode` argument? Seem like it's more complicated than that.

Comment (shashankg7) [inline keras/backend/theano_backend.py]: This op takes variance as argument (http://deeplearning.net/software/theano_versions/dev/library/sandbox/cuda/dnn.html#batch-normalization) and this method takes std as argument, that's why **2. I didn't at mode properly. I should have actually. 

batch normalization implementation in theano (deeplearning.net/software/theano/library/tensor/nnet/bn.html) is per-activation, right?

Comment (fchollet) [inline keras/backend/theano_backend.py]: It's irrelevant. Look at what the underlying implementation does with that
parameter and make sure that it matches the previous behavior. My guess is
that **2 should not be there.

Comment (shashankg7) [inline keras/backend/theano_backend.py]: Ok, I'll check with the underlying implementation and change accordingly.
